packages/grafana-o11y-ds-frontend/src/combineResponses.ts

```diff
@@ -159,9 +159,12 @@ function shouldCombine(frame1: DataFrame, frame2: DataFrame): boolean {
   // because we do not have a good "frametype" value for them yet.
   const customType1 = frame1.meta?.custom?.frameType;
   const customType2 = frame2.meta?.custom?.frameType;
-
+  // Legacy frames have this custom type
   if (customType1 === 'LabeledTimeValues' && customType2 === 'LabeledTimeValues') {
```

Comment:
See https://github.com/grafana/grafana/blob/main/pkg/tsdb/loki/frame.go#L141 and https://github.com/grafana/grafana/blob/main/pkg/tsdb/loki/frame.go#L209

Also this is very Loki specific and infinite scrolling is not Loki exclusive IIRC.